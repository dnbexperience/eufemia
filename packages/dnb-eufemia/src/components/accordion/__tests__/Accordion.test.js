/**
 * Component Test
 *
 */

import React from 'react'
import {
  mount,
  axeComponent,
  toJson,
  loadScss,
} from '../../../core/jest/jestSetup'
import Component from '../Accordion'

const props = {}
props.id = 'accordion'
props.variant = 'default'
props.no_animation = true
props.title = 'title'

describe('Accordion component', () => {
  // then test the state management
  const Comp = mount(<Component {...props} />)

  // mount compare the snapshot
  it('have to match snapshot', () => {
    expect(toJson(Comp)).toMatchSnapshot()
  })

  it('has correct state after "click" trigger', () => {
    // default expanded value has to be false
    expect(Comp.state().expanded).toBe(false)
    expect(
      Comp.find('.dnb-accordion__header')
        .instance()
        .getAttribute('aria-expanded')
    ).toBe('false')
    Comp.find('.dnb-accordion__header').simulate('click') // we could send inn the event data structure like this: , { target: { expanded: true } }
    expect(Comp.state().expanded).toBe(true)
    expect(
      Comp.find('.dnb-accordion__header')
        .instance()
        .getAttribute('aria-expanded')
    ).toBe('true')
    // Comp.find('.dnb-accordion__header').simulate('click')
    // expect(Comp.state().expanded).toBe(false)
    Comp.setProps({ expanded: false })
    expect(
      Comp.find('.dnb-accordion__header')
        .instance()
        .getAttribute('aria-expanded')
    ).toBe('false')
  })

  it('has "on_change" event which will trigger on click', () => {
    const my_event = jest.fn()
    const myEvent = jest.fn()
    const Comp = mount(
      <Component
        on_change={my_event}
        onChange={myEvent}
        expanded={false}
      />
    )
    // first click
    Comp.find('.dnb-accordion__header').simulate('click')
    expect(my_event).toHaveBeenCalled()
    expect(my_event.mock.calls[0][0].expanded).toBe(true)
    expect(myEvent.mock.calls.length).toBe(1)
    expect(myEvent.mock.calls[0][0]).toHaveProperty('expanded')
    expect(myEvent.mock.calls[0][0].expanded).toBe(true)
    // second click
    Comp.find('.dnb-accordion__header').simulate('click')
    expect(my_event.mock.calls[1][0].expanded).toBe(false)
  })

  it('has a disabled attribute, once we set disabled to true', () => {
    const Comp = mount(<Component />)
    Comp.setProps({
      disabled: true,
    })
    expect(
      Comp.find('.dnb-accordion__header')
        .instance()
        .hasAttribute('disabled')
    ).toBe(true)
  })

  it('should validate with ARIA rules', async () => {
    expect(await axeComponent(Comp)).toHaveNoViolations()
  })
})

describe('Accordion store API', () => {
  it('will save and read the states for a single accordion', () => {
    const inst = Component.Store('accordion-id')

    inst.saveState(true)
    expect(inst.getState()).toBe(true)
    expect(inst.getData()).toMatchObject({ expanded: true })

    inst.saveState(false)
    expect(inst.getState()).toBe(false)
    expect(inst.getData()).toMatchObject({ expanded: false })

    inst.flush()
    expect(inst.getState()).toBe(null)
    expect(inst.getData()).toBe(null)
  })

  it('will save and read the states for an accordion group', () => {
    const inst = Component.Group.Store('group-id')

    inst.saveState(true, 'remembered-state-2')
    expect(inst.getState('remembered-state-2')).toBe(true)
    expect(inst.getData()).toMatchObject({ id: 'remembered-state-2' })

    inst.saveState(false, 'remembered-state-2', { force: true })
    expect(inst.getState('remembered-state-2')).toBe(false)
    expect(inst.getData()).toMatchObject({ id: null })

    inst.flush('remembered-state-2')
    expect(inst.getState('remembered-state-2')).toBe(null)
    expect(inst.getData()).toBe(null)
  })
})

describe('Accordion group component', () => {
  const Comp = mount(
    <Component.Group label="Label" expanded id="group">
      <Component id="accordion-1" title="Accordion 1">
        Accordion 1
      </Component>
      <Component id="accordion-2" title="Accordion 2" expanded={false}>
        Accordion 2
      </Component>
    </Component.Group>
  )
  it('has to have correct aria-pressed', () => {
    expect(
      Comp.find('#accordion-1').exists('.dnb-accordion__content--hidden')
    ).toBe(false)
    expect(
      Comp.find('#accordion-2').exists('.dnb-accordion__content--hidden')
    ).toBe(true)
  })
  it('has "on_change" event which will trigger on a button click', () => {
    const my_event = jest.fn()
    const myEvent = jest.fn()
    const Comp = mount(
      <Component.Group
        id="group"
        on_change={my_event}
        onChange={myEvent}
        value="second"
        data-prop="group-value"
      >
        <Component
          id="accordion-1"
          text="Accordion 1"
          value="first"
          data-prop="value-1"
          attributes={{ 'data-attr': 'value' }}
        />
        <Component
          id="accordion-2"
          text="Accordion 2"
          value="second"
          data-prop="value-2"
          attributes={{ 'data-attr': 'value' }}
        />
      </Component.Group>
    )

    Comp.find('#accordion-1')
      .find('.dnb-accordion__header')
      .simulate('click')
    expect(my_event).toHaveBeenCalled()
    expect(my_event.mock.calls[0][0].id).toBe('accordion-1')
    expect(my_event.mock.calls[0][0].expanded).toBe(true)
    expect(myEvent.mock.calls.length).toBe(1)

    Comp.find('#accordion-2')
      .find('.dnb-accordion__header')
      .at(0)
      .simulate('click')
    expect(my_event.mock.calls[1][0].id).toBe('accordion-2')
    expect(my_event.mock.calls[1][0].expanded).toBe(true)

    Comp.find('#accordion-1')
      .find('.dnb-accordion__header')
      .at(0)
      .simulate('click')
    expect(my_event.mock.calls[2][0].expanded).toBe(true)
  })
})

describe('Accordion container component', () => {
  class DidRender extends React.PureComponent {
    state = { mounted: false }
    componentDidMount() {
      this.setState({ mounted: true })
    }
    render() {
      return <div id={this.props.id}>{String(this.state.mounted)}</div>
    }
  }

  const Increment = () => {
    const [count, setCount] = React.useState(1)
    return (
      <button id="increment" onClick={() => setCount((s) => s + 1)}>
        {count}
      </button>
    )
  }

  const Comp = mount(
    <>
      <Increment />
      <Component.Group
        label="Label"
        id="container"
        single_container
        prevent_rerender
        remember_state
      >
        <Component id="accordion-1" title="Accordion 1">
          Accordion 1
          <DidRender id="mounted-1" />
        </Component>
        <Component id="accordion-2" title="Accordion 2" expanded={true}>
          Accordion 2
          <DidRender id="mounted-2" />
        </Component>
        <Component id="accordion-3" title="Accordion 3">
          Accordion 3
          <DidRender id="mounted-3" />
        </Component>
      </Component.Group>
    </>
  )

  it('has only to render the expanded accordion content', () => {
    expect(Comp.find('button#increment').text()).toBe('1')
    Comp.find('button#increment').simulate('click')
    expect(Comp.find('button#increment').text()).toBe('2')

    expect(Comp.find('div#mounted-1').exists()).toBe(false)
    expect(Comp.find('div#mounted-2').text()).toBe('true')
    expect(Comp.find('div#mounted-3').exists()).toBe(false)
    expect(
      Comp.find('#accordion-2')
        .find('.dnb-accordion__header')
        .instance()
        .getAttribute('aria-expanded')
    ).toBe('true')

    Comp.find('#accordion-1')
      .find('.dnb-accordion__header')
      .simulate('click')

    Comp.find('button#increment').simulate('click')
    expect(Comp.find('button#increment').text()).toBe('3')

    expect(Comp.find('div#mounted-1').text()).toBe('true')
    expect(Comp.find('div#mounted-2').text()).toBe('true')
    expect(Comp.find('div#mounted-3').exists()).toBe(false)

    Comp.find('#accordion-2')
      .find('.dnb-accordion__header')
      .simulate('click')

    Comp.find('button#increment').simulate('click')
    expect(Comp.find('button#increment').text()).toBe('4')

    expect(Comp.find('div#mounted-3').exists()).toBe(false)

    Comp.find('#accordion-3')
      .find('.dnb-accordion__header')
      .simulate('click')

    Comp.find('button#increment').simulate('click')
    expect(Comp.find('button#increment').text()).toBe('5')

    expect(Comp.find('div#mounted-3').text()).toBe('true')
  })
})

describe('Accordion scss', () => {
  it('have to match snapshot', () => {
    const scss = loadScss(require.resolve('../style/dnb-accordion.scss'))
    expect(scss).toMatchSnapshot()
  })
  it('have to match default theme snapshot', () => {
    const scss = loadScss(
      require.resolve('../style/themes/dnb-accordion-theme-ui.scss')
    )
    expect(scss).toMatchSnapshot()
  })
})
