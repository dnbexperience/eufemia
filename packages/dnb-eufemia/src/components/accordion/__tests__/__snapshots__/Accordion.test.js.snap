// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion component have to match snapshot 1`] = `
<Accordion
  allow_close_all={null}
  attributes={null}
  class={null}
  className={null}
  custom_element={null}
  custom_method={null}
  disabled={null}
  element={null}
  expanded={null}
  expanded_ssr={null}
  flush_remembered_state={null}
  group={null}
  heading={null}
  heading_level={null}
  icon={null}
  icon_position={null}
  icon_size="medium"
  id="accordion"
  left_component={null}
  no_animation={true}
  on_change={null}
  on_state_update={null}
  prerender={null}
  prevent_rerender={null}
  prevent_rerender_conditional={null}
  remember_state={null}
  single_container={null}
  skeleton={null}
  title="title"
  variant="default"
>
  <div
    className="dnb-accordion dnb-accordion--no-animation dnb-accordion__variant--default"
    id="accordion"
  >
    <AccordionHeader
      className={null}
      description={null}
      disabled={null}
      element={null}
      heading={null}
      heading_level={null}
      icon={null}
      icon_position={null}
      icon_size="medium"
      id={null}
      left_component={null}
      skeleton={null}
      title={null}
    >
      <div
        aria-controls="accordion-content"
        aria-expanded={false}
        className="dnb-accordion__header"
        disabled={false}
        onClick={[Function]}
        onKeyDown={[Function]}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
        role="button"
        tabIndex="0"
      >
        <AccordionHeaderIcon
          expanded={false}
          icon={null}
          key="icon"
          size="medium"
        >
          <span
            className="dnb-accordion__header__icon"
          >
            <IconPrimary
              alt={null}
              aria-hidden={true}
              attributes={null}
              border={null}
              className={null}
              color={null}
              height={null}
              icon="chevron-down"
              inherit_color={true}
              modifier={null}
              size="medium"
              skeleton={null}
              title={null}
              width={null}
            >
              <span
                aria-hidden={true}
                className="dnb-icon dnb-icon--medium dnb-icon--inherit-color"
                data-testid="chevron-down icon"
                role="presentation"
              >
                <chevron_down_medium>
                  <svg
                    fill="none"
                    height={24}
                    viewBox="0 0 24 24"
                    width={24}
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="m19 8.5-7 7-7-7"
                      stroke="#000"
                      strokeLinecap="round"
                      strokeLinejoin="round"
                      strokeWidth={1.5}
                    />
                  </svg>
                </chevron_down_medium>
              </span>
            </IconPrimary>
          </span>
        </AccordionHeaderIcon>
        <AccordionHeaderContainer
          key="container"
        />
        <span
          className="dnb-accordion__header__wrapper"
          key="wrapper"
        >
          <AccordionHeaderTitle
            key="title"
          >
            <span
              className="dnb-accordion__header__title"
            >
              title
            </span>
          </AccordionHeaderTitle>
          <AccordionHeaderDescription
            key="description"
          />
        </span>
      </div>
    </AccordionHeader>
    <AccordionContent
      className={null}
      instance={null}
    >
      <div
        className="dnb-accordion__content dnb-accordion__content--hidden"
      >
        <div
          aria-disabled={true}
          aria-hidden={true}
          aria-labelledby="accordion-header"
          className="dnb-accordion__content__inner dnb-accordion__content__inner--remove-content"
          disabled={true}
          id="accordion-content"
          role="region"
        />
      </div>
    </AccordionContent>
  </div>
</Accordion>
`;

exports[`Accordion scss have to match default theme snapshot 1`] = `
"/*
* Accordion theme
*
*/
/**
 * This file is only used to make themes independent
 * so that they can get imported individually, without the core styles
 *
 */
/*
 * Utilities
 */
/*
* Button mixins
*
*/
.dnb-accordion__header {
  border: none; }

> .dnb-accordion__header .dnb-accordion__header__icon,
> .dnb-accordion__header .dnb-accordion__header__container {
  color: var(--color-emerald-green); }

> .dnb-accordion__header .dnb-accordion__header__description {
  color: var(--color-black-55); }

.dnb-accordion--expanded > .dnb-accordion__header .dnb-accordion__header__title {
  font-weight: var(--font-weight-medium); }

.dnb-accordion__variant--outlined > .dnb-accordion__header {
  color: var(--color-black-80);
  background-color: var(--color-white);
  --border-color: var(--color-black-8);
  box-shadow: inset 0 0 0 0.0625rem var(--border-color);
  /* iOS fix - because \\"inset\\" works not fine with border-radius */
  /* Safari fix - because \\"inset\\" works not fine with border-radius if the user zooms the page */
  border-color: transparent; }
  @supports (-webkit-touch-callout: none) {
    .dnb-accordion__variant--outlined > .dnb-accordion__header {
      box-shadow: 0 0 0 0.0625rem var(--border-color); } }
  @media not all and (min-resolution: 0.001dpcm) {
    @supports (-webkit-appearance: none) and (stroke-color: transparent) and (not (-webkit-overflow-scrolling: touch)) {
      .dnb-accordion__variant--outlined > .dnb-accordion__header {
        box-shadow: 0 0 0 0.0625rem var(--border-color); } } }
  @media screen and (-ms-high-contrast: none) {
    .dnb-accordion__variant--outlined > .dnb-accordion__header {
      box-shadow: inset 0 0 0 1px var(--color-black-8); } }
  .dnb-accordion__variant--outlined > .dnb-accordion__header:focus[disabled],
  html:not([data-whatintent='touch']) .dnb-accordion__variant--outlined > .dnb-accordion__header:focus[disabled] {
    cursor: not-allowed; }
  .dnb-accordion__variant--outlined > .dnb-accordion__header:focus:not([disabled]),
  html:not([data-whatintent='touch']) .dnb-accordion__variant--outlined > .dnb-accordion__header:focus:not([disabled]) {
    outline: none; }
    html[data-whatinput='keyboard'] .dnb-accordion__variant--outlined > .dnb-accordion__header:focus:not([disabled]), html[data-whatinput='keyboard']
    html:not([data-whatintent='touch']) .dnb-accordion__variant--outlined > .dnb-accordion__header:focus:not([disabled]) {
      color: var(--color-emerald-green);
      background-color: var(--color-white); }
    html[data-whatinput='keyboard'] .dnb-accordion__variant--outlined > .dnb-accordion__header:focus:not([disabled]), html[data-whatinput='keyboard']
    html:not([data-whatintent='touch']) .dnb-accordion__variant--outlined > .dnb-accordion__header:focus:not([disabled]) {
      --border-color: var(--color-emerald-green);
      box-shadow: inset 0 0 0 0.125rem var(--border-color);
      /* iOS fix - because \\"inset\\" works not fine with border-radius */
      /* Safari fix - because \\"inset\\" works not fine with border-radius if the user zooms the page */
      border-color: transparent; }
      @supports (-webkit-touch-callout: none) {
        html[data-whatinput='keyboard'] .dnb-accordion__variant--outlined > .dnb-accordion__header:focus:not([disabled]), html[data-whatinput='keyboard']
        html:not([data-whatintent='touch']) .dnb-accordion__variant--outlined > .dnb-accordion__header:focus:not([disabled]) {
          box-shadow: 0 0 0 0.125rem var(--border-color); } }
      @media not all and (min-resolution: 0.001dpcm) {
        @supports (-webkit-appearance: none) and (stroke-color: transparent) and (not (-webkit-overflow-scrolling: touch)) {
          html[data-whatinput='keyboard'] .dnb-accordion__variant--outlined > .dnb-accordion__header:focus:not([disabled]), html[data-whatinput='keyboard']
          html:not([data-whatintent='touch']) .dnb-accordion__variant--outlined > .dnb-accordion__header:focus:not([disabled]) {
            box-shadow: 0 0 0 0.125rem var(--border-color); } } }
      @media screen and (-ms-high-contrast: none) {
        html[data-whatinput='keyboard'] .dnb-accordion__variant--outlined > .dnb-accordion__header:focus:not([disabled]), html[data-whatinput='keyboard']
        html:not([data-whatintent='touch']) .dnb-accordion__variant--outlined > .dnb-accordion__header:focus:not([disabled]) {
          box-shadow: inset 0 0 0 0.125rem var(--color-emerald-green); } }
  html:not([data-whatintent='touch']) .dnb-accordion__variant--outlined > .dnb-accordion__header:hover[disabled] {
    cursor: not-allowed; }
  html:not([data-whatintent='touch']) .dnb-accordion__variant--outlined > .dnb-accordion__header:hover:not([disabled]) {
    color: var(--color-emerald-green);
    background-color: var(--color-white);
    --border-color: var(--color-emerald-green);
    box-shadow: 0 0 0 0.125rem var(--border-color);
    border-color: transparent; }
    @media screen and (-ms-high-contrast: none) {
      html:not([data-whatintent='touch']) .dnb-accordion__variant--outlined > .dnb-accordion__header:hover:not([disabled]) {
        box-shadow: 0 0 0 0.125rem var(--color-emerald-green); } }
    @media screen and (-ms-high-contrast: none) {
      html:not([data-whatintent='touch']) .dnb-accordion__variant--outlined > .dnb-accordion__header:hover:not([disabled]) {
        opacity: 1; } }
  .dnb-accordion__variant--outlined > .dnb-accordion__header:active[disabled],
  html:not([data-whatintent='touch']) .dnb-accordion__variant--outlined > .dnb-accordion__header:active[disabled] {
    cursor: not-allowed; }
  .dnb-accordion__variant--outlined > .dnb-accordion__header:active:not([disabled]),
  html:not([data-whatintent='touch']) .dnb-accordion__variant--outlined > .dnb-accordion__header:active:not([disabled]) {
    color: var(--color-emerald-green);
    background-color: var(--color-pistachio);
    --border-color: var(--color-emerald-green);
    box-shadow: 0 0 0 0.0625rem var(--border-color);
    border-color: transparent; }
    @media screen and (-ms-high-contrast: none) {
      .dnb-accordion__variant--outlined > .dnb-accordion__header:active:not([disabled]),
      html:not([data-whatintent='touch']) .dnb-accordion__variant--outlined > .dnb-accordion__header:active:not([disabled]) {
        box-shadow: 0 0 0 0.0625rem var(--color-emerald-green); } }

.dnb-accordion__variant--outlined > .dnb-accordion__header .dnb-accordion__header[disabled] {
  background-color: var(--color-white);
  --border-color: var(--color-sea-green-30);
  box-shadow: 0 0 0 0.0625rem var(--border-color);
  border-color: transparent; }
  @media screen and (-ms-high-contrast: none) {
    .dnb-accordion__variant--outlined > .dnb-accordion__header .dnb-accordion__header[disabled] {
      box-shadow: 0 0 0 0.0625rem var(--color-sea-green-30); } }
  .dnb-accordion__variant--outlined > .dnb-accordion__header .dnb-accordion__header[disabled] * {
    color: var(--color-sea-green-30); }

.dnb-accordion__variant--outlined.dnb-accordion--expanded > .dnb-accordion__header {
  color: var(--color-white);
  background-color: var(--color-sea-green);
  --border-color: transparent;
  box-shadow: 0 0 0 0.0625rem var(--border-color);
  border-color: transparent; }
  @media screen and (-ms-high-contrast: none) {
    .dnb-accordion__variant--outlined.dnb-accordion--expanded > .dnb-accordion__header {
      box-shadow: 0 0 0 0.0625rem transparent; } }

.dnb-accordion__variant--outlined.dnb-accordion--expanded > .dnb-accordion__header .dnb-accordion__header[disabled] {
  background-color: var(--color-sea-green-30); }
  .dnb-accordion__variant--outlined.dnb-accordion--expanded > .dnb-accordion__header .dnb-accordion__header[disabled] * {
    color: var(--color-white); }

.dnb-accordion__variant--outlined.dnb-accordion--expanded > .dnb-accordion__header html[data-whatinput='keyboard']
.dnb-accordion__header:not([disabled]):not(:active):not(:hover):focus {
  background-color: var(--color-mint-green); }
  .dnb-accordion__variant--outlined.dnb-accordion--expanded > .dnb-accordion__header html[data-whatinput='keyboard']
.dnb-accordion__header:not([disabled]):not(:active):not(:hover):focus * {
    color: var(--color-emerald-green); }

.dnb-accordion__variant--outlined.dnb-accordion--expanded > .dnb-accordion__header .dnb-accordion__header:not([disabled]):not(:active):not(:hover):focus {
  outline: none; }
  html[data-whatinput='keyboard'] .dnb-accordion__variant--outlined.dnb-accordion--expanded > .dnb-accordion__header .dnb-accordion__header:not([disabled]):not(:active):not(:hover):focus {
    --border-color: var(--color-emerald-green);
    box-shadow: inset 0 0 0 0.125rem var(--border-color);
    /* iOS fix - because \\"inset\\" works not fine with border-radius */
    /* Safari fix - because \\"inset\\" works not fine with border-radius if the user zooms the page */
    border-color: transparent; }
    @supports (-webkit-touch-callout: none) {
      html[data-whatinput='keyboard'] .dnb-accordion__variant--outlined.dnb-accordion--expanded > .dnb-accordion__header .dnb-accordion__header:not([disabled]):not(:active):not(:hover):focus {
        box-shadow: 0 0 0 0.125rem var(--border-color); } }
    @media not all and (min-resolution: 0.001dpcm) {
      @supports (-webkit-appearance: none) and (stroke-color: transparent) and (not (-webkit-overflow-scrolling: touch)) {
        html[data-whatinput='keyboard'] .dnb-accordion__variant--outlined.dnb-accordion--expanded > .dnb-accordion__header .dnb-accordion__header:not([disabled]):not(:active):not(:hover):focus {
          box-shadow: 0 0 0 0.125rem var(--border-color); } } }
    @media screen and (-ms-high-contrast: none) {
      html[data-whatinput='keyboard'] .dnb-accordion__variant--outlined.dnb-accordion--expanded > .dnb-accordion__header .dnb-accordion__header:not([disabled]):not(:active):not(:hover):focus {
        box-shadow: inset 0 0 0 0.125rem var(--color-emerald-green); } }

.dnb-accordion__variant--outlined.dnb-accordion--expanded > .dnb-accordion__header .dnb-accordion__header:not(:hover):not([disabled]) * {
  color: var(--color-white); }

html:not([data-whatintent='touch']) .dnb-accordion__variant--outlined.dnb-accordion--expanded > .dnb-accordion__header.dnb-accordion--hover:not(:active):hover[disabled] {
  cursor: not-allowed; }

html:not([data-whatintent='touch']) .dnb-accordion__variant--outlined.dnb-accordion--expanded > .dnb-accordion__header.dnb-accordion--hover:not(:active):hover:not([disabled]) {
  color: var(--color-white);
  background-color: var(--color-sea-green);
  --border-color: var(--color-sea-green);
  box-shadow: 0 0 0 0.0625rem var(--border-color);
  border-color: transparent; }
  @media screen and (-ms-high-contrast: none) {
    html:not([data-whatintent='touch']) .dnb-accordion__variant--outlined.dnb-accordion--expanded > .dnb-accordion__header.dnb-accordion--hover:not(:active):hover:not([disabled]) {
      box-shadow: 0 0 0 0.0625rem var(--color-sea-green); } }

.dnb-accordion__variant--outlined.dnb-accordion--expanded > .dnb-accordion__header.dnb-accordion--hover:not(:active) * {
  color: var(--color-white); }
"
`;

exports[`Accordion scss have to match snapshot 1`] = `
"/*
* DNB Accordion
*
*/
/**
 * This file is only used to make components independent
 * so that they can get imported individually, without the core styles
 *
 */
/*
 * Utilities
 */
/*
 * Scopes
 *
 */
/*
 * Document Reset
 *
 */
/*
* DNB icon
*
*/
/*
* Icon component
*
*/
.dnb-icon {
  display: inline-block;
  vertical-align: middle;
  font-size: 1rem;
  line-height: 1rem;
  color: inherit;
  width: 1em;
  height: 1em; }
  .dnb-icon img,
  .dnb-icon svg {
    width: inherit;
    height: inherit;
    shape-rendering: geometricPrecision;
    vertical-align: top; }
  .dnb-icon svg[width='100%'] {
    width: inherit; }
  .dnb-icon svg[height='100%'] {
    height: inherit; }
  .dnb-icon--inherit-color svg:not([fill]),
  .dnb-icon--inherit-color svg [fill] {
    fill: currentColor; }
  .dnb-icon--inherit-color svg [stroke] {
    stroke: currentColor; }
  .dnb-icon--small {
    font-size: 0.75rem; }
  .dnb-icon--default {
    font-size: 1rem; }
  .dnb-icon--medium {
    font-size: 1.5rem; }
  .dnb-icon--large {
    font-size: 2rem; }
  .dnb-icon--x-large {
    font-size: 2.5rem; }
  .dnb-icon--xx-large {
    font-size: 3rem; }
  .dnb-icon--custom-size {
    width: auto;
    height: auto; }
  .dnb-icon--auto {
    font-size: 1em; }
  .dnb-icon--auto > .dnb-icon--wrapper {
    display: inline-flex;
    align-items: center;
    justify-content: center; }
  h1 .dnb-icon,
  h2 .dnb-icon,
  h3 .dnb-icon,
  h4 .dnb-icon,
  h5 .dnb-icon,
  h6 .dnb-icon {
    vertical-align: middle; }
  .dnb-icon.dnb-skeleton {
    color: var(--skeleton-color) !important; }
    .dnb-icon.dnb-skeleton::before, .dnb-icon.dnb-skeleton::after {
      content: none !important; }
  @media screen and (-ms-high-contrast: none) {
    .dnb-icon {
      flex: none; } }

.dnb-accordion {
  font-family: var(--font-family-default);
  font-weight: var(--font-weight-basis);
  font-size: var(--font-size-small);
  font-style: normal;
  line-height: var(--line-height-basis);
  color: var(--color-black-80, #333);
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
  -webkit-tap-highlight-color: rgba(0, 0, 0, 0);
  /**
  * Ensure consistency and use the same as HTML reset -> html {...}
  * between base and code package
  */
  -moz-tab-size: 4;
  tab-size: 4;
  -ms-text-size-adjust: 100%;
  -webkit-text-size-adjust: 100%;
  word-break: break-word;
  /**
  * 1. Remove repeating backgrounds in all browsers (opinionated).
  * 2. Add border box sizing in all browsers (opinionated).
  */
  /**
  * 1. Add text decoration inheritance in all browsers (opinionated).
  * 2. Add vertical alignment inheritance in all browsers (opinionated).
  */
  margin: 0;
  padding: 0; }
  .dnb-accordion *,
  .dnb-accordion ::before,
  .dnb-accordion ::after {
    background-repeat: no-repeat;
    /* 1 */
    box-sizing: border-box;
    /* 2 */ }
  .dnb-accordion ::before,
  .dnb-accordion ::after {
    text-decoration: inherit;
    /* 1 */
    vertical-align: inherit;
    /* 2 */ }

/*
* Accordion component
*
*/
:root {
  --accordion-border-width: 0.0625rem;
  --accordion-border-radius: 0.25rem;
  --accordion-easing: cubic-bezier(0.42, 0, 0, 1); }

.dnb-accordion {
  position: relative;
  display: flex;
  flex-direction: column;
  font-size: var(--font-size-small);
  line-height: var(--line-height-basis); }
  .dnb-accordion__header {
    display: flex;
    align-items: center;
    cursor: pointer;
    outline: none;
    border: var(--accordion-border-width) solid transparent;
    border-radius: var(--accordion-border-radius); }
    .dnb-accordion__header:focus {
      user-select: none; }
    .dnb-accordion__header * {
      pointer-events: none; }
    .dnb-accordion__header--prevent-click {
      user-select: none;
      pointer-events: none; }
    .dnb-accordion__header__wrapper {
      display: flex;
      flex-direction: column;
      width: 100%;
      margin: 1.5rem 0 1.5rem 2rem; }
    .dnb-accordion__header--description .dnb-accordion__header__wrapper {
      margin: 1rem 0 1rem 2rem; }
    .dnb-accordion__header__icon + .dnb-accordion__header__wrapper {
      margin-left: 0;
      margin-right: 2rem; }
    .dnb-accordion__header__container {
      margin: 0 1rem; }
    .dnb-accordion__header__icon {
      align-self: flex-start;
      margin: 1.5rem 1rem;
      font-size: var(--font-size-small);
      line-height: var(--line-height-small);
      transition: transform 400ms var(--accordion-easing); }
    .dnb-accordion__header--description .dnb-accordion__header__icon {
      margin-top: 1rem; }
    .dnb-accordion__header__icon--right {
      justify-content: space-between; }
    .dnb-accordion__header__icon--right .dnb-accordion__header__icon {
      order: 3; }
    .dnb-accordion__header__icon--right .dnb-accordion__header__container {
      order: 1; }
    .dnb-accordion__header__icon--right .dnb-accordion__header__wrapper {
      order: 2;
      margin-left: 1rem;
      margin-right: 1rem; }
    .dnb-accordion__header__icon--right .dnb-accordion__header__wrapper + .dnb-accordion__header__container {
      margin-right: 0; }
    .dnb-accordion__header__icon--right .dnb-accordion__header__container + .dnb-accordion__header__wrapper {
      margin-left: 0; }
    .dnb-accordion__header__title {
      font-size: var(--font-size-basis);
      line-height: var(--line-height-basis); }
    .dnb-accordion__header__description {
      font-size: var(--font-size-small);
      line-height: var(--line-height-small);
      font-weight: var(--font-weight-basis); }
    .dnb-accordion__header__title + .dnb-accordion__header__description {
      margin-top: 0.25rem; }
    .dnb-accordion__header__description + .dnb-accordion__header__title {
      margin-top: 0.25rem; }
  .dnb-accordion--no-animation .dnb-accordion__header__icon,
  html[data-visual-test] .dnb-accordion__header__icon {
    transition-duration: 1ms !important; }
  .dnb-accordion--expanded > .dnb-accordion__header .dnb-accordion__header__icon {
    transform: rotate(-180deg); }
  .dnb-accordion__content {
    display: flex;
    opacity: 1;
    visibility: visible;
    width: 100%;
    will-change: height;
    transition: height 400ms var(--accordion-easing), opacity 600ms var(--accordion-easing); }
    .dnb-accordion__content--hidden {
      visibility: hidden;
      height: 0;
      opacity: 0; }
    .dnb-accordion__content--is-animating {
      overflow: hidden; }
    .dnb-accordion__content__inner {
      width: 100%;
      margin-top: 1rem; }
      .dnb-accordion__content__inner--remove-content {
        display: none; }
  .dnb-accordion-group--single-container {
    transition: min-height 1s var(--accordion-easing); }
  .dnb-accordion-group--single-container .dnb-accordion {
    position: static;
    max-width: 60rem; }
    @media screen and (min-width: 40em) {
      .dnb-accordion-group--single-container .dnb-accordion__header {
        width: 40%; }
      .dnb-accordion-group--single-container .dnb-accordion__content {
        margin-left: 1rem;
        transition: opacity 1s var(--accordion-easing); }
        .dnb-accordion-group--single-container .dnb-accordion__content__inner {
          margin-top: 0; } }
  @media screen and (min-width: 40em) {
    .dnb-accordion-group--single-container .dnb-accordion > .dnb-accordion__header .dnb-accordion__header__icon {
      transform: rotate(-90deg); } }
  .dnb-accordion--no-animation .dnb-accordion__content,
  html[data-visual-test] .dnb-accordion__content {
    transition-duration: 1ms !important; }
  .dnb-accordion-group--single-container .dnb-accordion-group__children {
    max-width: 60rem; }
    @media screen and (min-width: 40em) {
      .dnb-accordion-group--single-container .dnb-accordion-group__children {
        position: relative;
        display: flex;
        flex-direction: column; }
        .dnb-accordion-group--single-container .dnb-accordion-group__children .dnb-accordion__content {
          position: absolute;
          z-index: 10;
          top: 0;
          right: 0;
          width: 60%; } }
"
`;
